src/python/pants/backend/python/util_rules/interpreter_constraints.py

```diff
@@ -35,12 +35,18 @@ def interpreter_constraints(self) -> InterpreterConstraintsField:
 _FS = TypeVar("_FS", bound=FieldSetWithInterpreterConstraints)
 
 
-# The current maxes are 2.7.18 and 3.6.13.
-_EXPECTED_LAST_PATCH_VERSION = 18
+# The current maxes are 2.7.18 and 3.6.15.  We go much higher, for safety.
+_PATCH_VERSION_UPPER_BOUND = 30
 
 
 # Normally we would subclass `DeduplicatedCollection`, but we want a custom constructor.
 class InterpreterConstraints(FrozenOrderedSet[Requirement], EngineAwareParameter):
+    @classmethod
+    def for_fixed_python_version(
+        cls, cpython_version_str: str, interpreter_type: str = "CPython"
```

Comment:
The first arg name doesn't make sense if the `interpreter_type` is overridden. Perhaps just `version_str`?

Reply:
Fixed.